lib/iris/experimental/ugrid/mesh.py

```diff
@@ -975,77 +925,129 @@ def __ne__(self, other):
-        if edge_coords_and_axes:
-            args.append(f"edge_coords_and_axes={edge_coords_and_axes}")
-
-        # optional argument
-        if self.topology_dimension > 1:
-            face_coords_and_axes = to_coord_and_axis(self.face_coords)
-            if face_coords_and_axes:
-                args.append(f"face_coords_and_axes={face_coords_and_axes}")
-
-        def kwargs_filter(k, v):
-            result = False
-            if k != "topology_dimension":
-                if not (
-                    self.topology_dimension == 1 and k == "face_dimension"
-                ):
-                    if v is not None:
-                        result = True
-                        if (
-                            not isinstance(v, str)
-                            and isinstance(v, Iterable)
-                            and not v
-                        ):
-                            result = False
-                        elif k == "units" and v == "unknown":
-                            result = False
-            return result
-
-        # optional arguments (metadata)
-        args.extend(
-            [
-                f"{k}={v!r}"
-                for k, v in self.metadata._asdict().items()
-                if kwargs_filter(k, v)
-            ]
+    def __str__(self):
+        return self.summary(shorten=False)
+
+    def _summary_oneline(self):
+        # We use the repr output to produce short one-line identity summary,
+        # similar to the object.__str__ output "<object at xxx>".
+        # This form also used in other str() constructions, like MeshCoord.
+        # By contrast, __str__ (below) produces a readable multi-line printout.
+        mesh_name = self.name()
+        if mesh_name in (None, "", "unknown"):
+            mesh_name = None
+        if mesh_name:
+            # Use a more human-readable form
+            mesh_string = f"<Mesh: '{mesh_name}'>"
+        else:
+            # Mimic the generic object.__str__ style.
+            mesh_id = id(self)
+            mesh_string = f"<Mesh object at {hex(mesh_id)}>"
+
+        return mesh_string
+
+    def _summary_multiline(self):
+        # Produce a readable multi-line summary of the Mesh content.
+        lines = []
+        n_indent = 4
+        indent_str = " " * n_indent
+
+        def line(text, i_indent=0):
+            indent = indent_str * i_indent
+            lines.append(f"{indent}{text}")
+
+        line(f"Mesh : '{self.name()}'")
+        line(f"topology_dimension: {self.topology_dimension}", 1)
+        for element in ("node", "edge", "face"):
+            if element == "node":
+                element_exists = True
+            else:
+                main_conn_name = f"{element}_node_connectivity"
+                main_conn = getattr(self, main_conn_name, None)
+                element_exists = main_conn is not None
+            if element_exists:
+                # Include a section for this element
+                line(element, 1)
```

Comment:
```suggestion
                line(f"{element}:", 1)
```

Reply:
TBH I'm not too sure about this.  I considered + rejected this, initially, because I'm concerned about having colons "everywhere".
Since there  are effectively sections + sub-sections here (e.g. "node coordinates"), and the individual coords + connectivities also use colons within the reprs, we would wind up with a whole lot of them.

If we do do this, I think we should be consistent and add it into the other 'subsection' headings.
BTW the [re-format of the attributes listing](https://github.com/SciTools/iris/discussions/4501#discussioncomment-2034628) will affect this as well.  I'm just going to use a best-guess for that here ..

Trying it out, I also think that 'spacing off' the colon makes sense, when the following content is below and not on the same line
For example:
```
Mesh : 'my_topology_mesh'
    topology_dimension: 2
    node :
        node_dimension: 'NodeDim'
        node coordinates :
            <AuxCoord: longitude / (unknown)  [...]  shape(3,)>
            <AuxCoord: latitude / (unknown)  [...]  shape(3,)>
    face :
        face_dimension: 'FaceDim'
        face_node_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face coordinates :
            <AuxCoord: longitude / (unknown)  [...]>
            <AuxCoord: latitude / (unknown)  [...]>
    optional connectivities :
        face_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face_edge_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        edge_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(3, 2)>
    long_name: 'my_topology_mesh'
    var_name: 'mesh'
    attributes :
        notes        'this is a test'
        history      'created in my model'
        version      2.3
        Conventions  'CF 1.9; UGRID 1.0'
```

Well... it's not rubbish, but I'm not sure it helps any.
On the other hand, is is probably more consistent with what we now have in the Cube printout
E.G.
```
unknown / (unknown)                 (-- : 1)
    Auxiliary coordinates:
        my_aux                          x
    Attributes:
        dict                        {'a': 1, 'b': 'two'}
        float                       3.2
        int                         9
        string key                  this info
```

So.. there we currently have a colon (not spaced-off) for each 'section title', 
and no colons between key + value in the attributes
But there are no 'subsections' in that case, which is maybe what makes the above look a bit messy ?

Reply:
P.S. Now I've finally committed that, I'm not sure about the 'spacing' off' either.
Here's the above without that ..
```
Mesh : 'my_topology_mesh'
    topology_dimension: 2
    node:
        node_dimension: 'NodeDim'
        node coordinates:
            <AuxCoord: longitude / (unknown)  [...]  shape(3,)>
            <AuxCoord: latitude / (unknown)  [...]  shape(3,)>
    face:
        face_dimension: 'FaceDim'
        face_node_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face coordinates:
            <AuxCoord: longitude / (unknown)  [...]>
            <AuxCoord: latitude / (unknown)  [...]>
    optional connectivities:
        face_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face_edge_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        edge_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(3, 2)>
    long_name: 'my_topology_mesh'
    var_name: 'mesh'
    attributes:
        notes        'this is a test'
        history      'created in my model'
        version      2.3
        Conventions  'CF 1.9; UGRID 1.0'
```
.. and here with no section colons at all..
```
Mesh : 'my_topology_mesh'
    topology_dimension: 2
    node
        node_dimension: 'NodeDim'
        node coordinates
            <AuxCoord: longitude / (unknown)  [...]  shape(3,)>
            <AuxCoord: latitude / (unknown)  [...]  shape(3,)>
    face
        face_dimension: 'FaceDim'
        face_node_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face coordinates
            <AuxCoord: longitude / (unknown)  [...]>
            <AuxCoord: latitude / (unknown)  [...]>
    optional connectivities
        face_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        face_edge_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(1, 3)>
        edge_face_connectivity: <Connectivity: unknown / (unknown)  [...]  shape(3, 2)>
    long_name: 'my_topology_mesh'
    var_name: 'mesh'
    attributes
        notes        'this is a test'
        history      'created in my model'
        version      2.3
        Conventions  'CF 1.9; UGRID 1.0'
```

I'm quite conflicted on this now

Reply:
I think I'm coming around to your original format, with no colons. Describing it as sections and subsections helps. I was coming at it from the perspective of what a regular coordinate prints out, but this is slightly different.

I'm happy to stick with how you have it